src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -460,7 +467,24 @@ private static bool StatementClosingDelimiterIsMissing(SyntaxNode currentNode)
         private static bool RequiredDelimiterIsMissing(SyntaxNode currentNode)
         {
             return currentNode.GetBrackets().closeBracket.IsMissing ||
-                currentNode.GetParentheses().closeParen.IsMissing;
+                currentNode.GetParentheses().closeParen.IsMissing ||
+                currentNode.GetBraces().closeBrace.IsMissing;
+        }
+
+        private static bool TryGetParentsWithBraces(SyntaxNode currentNode, [NotNullWhen(true)] out SyntaxNode? parentsWithBraces)
+        {
+            // Switch Expression could be followed by semicolon.
+            if (currentNode.GetAncestorsOrThis(IsSyntaxNodeWithBracesCouldBeFollowedBySemicolon).FirstOrDefault() is { } parent)
+            {
+                parentsWithBraces = parent;
+                return true;
+            }
+
+            parentsWithBraces = null;
+            return false;
         }
+
+        private static bool IsSyntaxNodeWithBracesCouldBeFollowedBySemicolon(SyntaxNode node)
```

Comment:
object/collection initializers?

Reply:
I do test initializers (if you see the comment history)
However, there are many tests failing, like https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.EditorFeatures.UnitTests/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs,1704
So I decided to start with switch expression only